### PR TITLE
initial simple server that serves up the build directory

### PR DESCRIPTION
**Pull request by benadida**
minimal. For now just serves the static files appropriately, with simple instructions.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,12 +1,13 @@
 version: 2
 jobs:
-  build:
+  test-frontend-and-server:
     docker:
-      - image: cypress/browsers:chrome67
+      #      - image: cypress/browsers:chrome67
+      - image: circleci/python:3.6.7-node-browsers
```
Comment (beausmith) [inline .circleci/config.yml]: nit: Does this image contain all browsers in commented line above. If so, delete comment.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
     steps:
       - checkout
       - restore_cache: # special step to restore the dependency cache
-          key: dotcache-cache-{{ checksum "package.json" }}
+          key: dotcache-cache-v20190430-{{ checksum "package.json" }}
```
Comment (beausmith) [inline .circleci/config.yml]: Curious: Why the hard-coded version here and below?

Also: When do we update this hard-coded version?

File: SERVER.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+The instructions here are for Ubuntu Linux 18.04
+
+## Install Requisite Software
+
+```
+sudo apt install python3
+pip3 install pipenv
+cd server/
+pipenv install
```
[the remark this answers is not in the document]
Comment (beausmith) [reply SERVER.md]: I backed up and tried a different route…

`brew update python`

Add `export PATH=/usr/local/opt/python/libexec/bin:$PATH` to my bash env

Let's try again…

First version checks…

```
beau at cleaver in ~/Development/VotingWorks/bmd on add-python-server [$]
$ python --version
Python 3.7.3

beau at cleaver in ~/Development/VotingWorks/bmd on add-python-server [$]
$ pip --version
pip 19.0.3 from /usr/local/lib/python3.7/site-packages/pip (python 3.7)

beau at cleaver in ~/Development/VotingWorks/bmd on add-python-server [$]
$ pipenv --version
pipenv, version 2018.11.26
```

Cool. Let's install…

```
beau at cleaver in ~/Development/VotingWorks/bmd on add-python-server [$]
$ cd server

beau at cleaver in ~/Development/VotingWorks/bmd/server on add-python-server [$]
$ pipenv install
Warning: Python 3.6 was not found on your system…
You can specify specific versions of Python with:
  $ pipenv --python path/to/python
```

Hmm…?  Do I need older version of Python? Let's try specifying python version…

```
beau at cleaver in ~/Development/VotingWorks/bmd/server on add-python-server [$]
$ which python
/usr/local/opt/python/libexec/bin/python

beau at cleaver in ~/Development/VotingWorks/bmd/server on add-python-server [$]
$ pipenv --python /usr/local/opt/python/libexec/bin/python
Creating a virtualenv for this project…
Pipfile: /Users/beau/Development/VotingWorks/bmd/server/Pipfile
Using /usr/local/opt/python/libexec/bin/python (3.7.3) to create virtualenv…
⠦ Creating virtual environment...Using base prefix '/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7'
New python executable in /Users/beau/.local/share/virtualenvs/server-pgI_f351/bin/python3.7
Also creating executable in /Users/beau/.local/share/virtualenvs/server-pgI_f351/bin/python
Installing setuptools, pip, wheel...
done.
Running virtualenv with interpreter /usr/local/opt/python/libexec/bin/python

✔ Successfully created virtual environment! 
Virtualenv location: /Users/beau/.local/share/virtualenvs/server-pgI_f351
Warning: Your Pipfile requires python_version 3.6, but you are using 3.7.3 (/Users/beau/.local/share/v/s/bin/python).
  $ pipenv --rm and rebuilding the virtual environment may resolve the issue.
  $ pipenv check will surely fail.
```

Looks like it worked, but `pipenv check will surely fail.` 😟 

Proceeding…

```
beau at cleaver in ~/Development/VotingWorks/bmd/server on add-python-server [$]
$ pipenv install
Warning: Your Pipfile requires python_version 3.6, but you are using 3.7.3 (/Users/beau/.local/share/v/s/bin/python).
  $ pipenv --rm and rebuilding the virtual environment may resolve the issue.
  $ pipenv check will surely fail.
Installing dependencies from Pipfile.lock (41bd8f)…
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 15/15 — 00:00:03
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.
```

Nope. Failed.

```
beau at cleaver in ~/Development/VotingWorks/bmd/server on add-python-server [$]
$ pipenv shell
Warning: Your Pipfile requires python_version 3.6, but you are using 3.7.3 (/Users/beau/.local/share/v/s/bin/python).
  $ pipenv --rm and rebuilding the virtual environment may resolve the issue.
  $ pipenv check will surely fail.
Launching subshell in virtual environment…
bash: prompt_git: command not found

beau at cleaver in ~/Development/VotingWorks/bmd/server
$  . /Users/beau/.local/share/virtualenvs/server-pgI_f351/bin/activate
bash: prompt_git: command not found
(server) 
```

Should I install older version of python (`3.6`) locally… or should we upgrade the project to latest python 3.7.3?

File: server/tests/test_server.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    app.config['TESTING'] = True
+    client = app.test_client()
+
+    yield client
+
+    # any cleanup goes here
+
+def test_index(client):
+    index_page = client.get("/")
+    assert b'VotingWorks Ballot Marking Device' in index_page.data
```
Comment (beausmith) [inline server/tests/test_server.py]: 😄

File: SERVER.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+When running on an actual voting machine in localhost mode, you should use this
+simple server.
+
+The instructions here are for Ubuntu Linux 18.04
+
+## Install Requisite Software
+
+```
+sudo apt install python3
```
Comment (beausmith) [inline SERVER.md]: nit: When sudo is used, I've been conditioned to always look for a less forceful path.

File: .circleci/config.yml [hunk 6]
```diff
@@ -16,12 +17,29 @@ jobs:
...
+          name: Run Front-End Test and Coverage
           command: |
             ./cc-test-reporter before-build
             yarn test
             ./cc-test-reporter after-build --exit-code $?
+      - run:
+          name: Set up python tools for backend
+          command: |
+            cd server
+            sudo pip install pipenv
```
[the remark this answers is not in the document]
Comment (benadida) [reply .circleci/config.yml]: good call, I'll check.

File: server/Makefile [hunk 7]
```diff
@@ -0,0 +1,9 @@
+
+test:
+	python -m pytest
+
+coverage:
+	python -m pytest --cov=bmdserver --cov-report term-missing --cov-fail-under=100 tests/
+
+run:
+	FLASK_APP=bmdserver.core python -m flask run --port 3000
```
[the remark this answers is not in the document]
Comment (benadida) [reply server/Makefile]: I'll check :) I'm not a `Make` expert, so this is emacs default

File: .circleci/config.yml (hunk 1 above)
Comment (benadida) [reply .circleci/config.yml]: yes it does, I'll delete.

File: .circleci/config.yml (hunk 2 above)
Comment (benadida) [reply .circleci/config.yml]: had to update it cause image change required it. I thought date would be a good way to tag it. I can add a comment?

File: server/Makefile (hunk 7 above)
[the remark this answers is not in the document]
Comment (benadida) [reply server/Makefile]: looks like, unless we use GNU Make and make everyone else use it, tabs it is:
https://stackoverflow.com/questions/2131213/can-you-make-valid-makefiles-without-tab-characters

File: SERVER.md (hunk 5 above)
Comment (benadida) [reply SERVER.md]: it's a little tricky to not use `sudo` in this particular case. I'll keep an eye out for how to find an alternative to this in the future.

File: SERVER.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (benadida) [reply SERVER.md]: I'm upgrading to Python 3.7.3 to make this a bit easier, but I don't think including Mac local environment is an important goal right now unless it's relatively easy. By all means if you get it to work with small tweaks, let's PR it!

File: server/Makefile (hunk 7 above)
[the remark this answers is not in the document]
Comment (case) [reply server/Makefile]: 💯

File: .circleci/config.yml (hunk 2 above)
Comment (beausmith) [reply .circleci/config.yml]: Ideally solution ✨ automagical 🦄 caching. 

There are many things we can use to create a unique key:
- [Keys and Templates](https://circleci.com/docs/2.0/caching/#using-keys-and-templates)
- [Built-in Environment Variables](https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables)

My suggestion:
```
key: bmd-cache-{{checksum ".circleci/config.yml"}}-{{ checksum "package.json" }}
```

Comment (beausmith) [reply .circleci/config.yml]: Tracking here: https://github.com/votingworks/bmd/issues/285

**Pull request merged**